DigitalLearningSolutions.Data.Tests/Services/RegistrationServiceTests.cs

```diff
@@ -674,7 +673,80 @@ public void PromoteDelegateToAdmin_throws_email_in_use_AdminCreationFailedExcept
             );
 
             // Then
-            result.Error.Should().Be(AdminCreationError.EmailAlreadyInUse);
+            using (new AssertionScope())
+            {
+                UpdateToExistingAdminAccountMustNotHaveHappened();
+                result.Error.Should().Be(AdminCreationError.EmailAlreadyInUse);
+            }
+        }
+
+        [Test]
+        public void
+            PromoteDelegateToAdmin_throws_email_in_use_AdminCreationFailedException_if_inactive_admin_at_different_centre_already_exists()

```

Comment:
I'm a bit confused by this. Is there a business requirement which states that if you are an inactive admin at any centre, you can't be made an active admin at a different centre?

Reply:
Pre-UAR, the AdminUsers table had a unique email constraint, so if you try to promote a delegate at your centre with an email address matching an admin at a different centre, it is not possible. With the changes here to prevent errors when attempting to do this, we want to allow reactivating the admin at the current centre, but not allow reactivating an admin at a different centre.
To get around this behaviour, they would probably have to change the email of the delegate + then promote. This would make them have the pair of accounts at the same correct centre. (The old account system is pretty confusing/bad)